### PR TITLE
init: Output value for sys.path in config

### PR DESCRIPTION
**Pull request by wenzeslaus**
With this addition, running `grass --config python_path` outputs the value needed for sys.path before the grass package can be imported.

Instead of having to hardcode the `etc/python` part of the path like this:

```python
gisbase = subprocess.check_output(["grass", "--config", "path"], text=True).strip()
sys.path.append(os.path.join(gisbase, "etc", "python"))
```

the new parameter allows user to (optionally) use more general code:

```python
python_path = subprocess.check_output(["grass", "--config", "python_path"], text=True).strip()
sys.path.append(python_path)
```

or alternatively in one call without a variable:

```python
sys.path.append(
    subprocess.check_output(["grass", "--config", "python_path"], text=True).strip()
)
```

Taking advantage of #1838, GISBASE environmental variable does not have to be set up before for importing _grass.script_ package, so setting up _sys.path_ is the only step user needs to do before that. Additionally, with #1829 (and #1979) the installation path (GISBASE) can be determined automatically from the path to the _grass.script.setup_ module (file) or the executable (if it is called _grass_).

Test online:
1. [binder link: plain grass.script setup](https://mybinder.org/v2/gh/wenzeslaus/grass/python-sys-path-from-grass-exec?urlpath=lab%2Ftree%2Fdoc%2Fnotebooks%2Fbasic_example.ipynb)
2. [binder link: grass.jupyter setup](https://mybinder.org/v2/gh/wenzeslaus/grass/python-sys-path-from-grass-exec?urlpath=lab%2Ftree%2Fdoc%2Fnotebooks%2Fbasic_example_grass_jupyter.ipynb)

### CODE REVIEW

File: lib/init/grass.py [hunk 1]
```diff
@@ -2145,6 +2154,8 @@ def print_params(params):
     for arg in params:
         if arg == "path":
             sys.stdout.write("%s\n" % GISBASE)
+        elif arg in ["python_path" , "python-path"]:
```
[the remark this answers is not in the document]
Comment (wenzeslaus) [reply lib/init/grass.py]: Thanks. Just forgot to run Black before commit/push.

**Comment by wenzeslaus** [comment]
@chaedri please have a look at the notebooks whether the code and description make sense and are at least somewhat understandable.

**Comment by veroandreo** [comment]
I don't know if related, but I get the following error in the last cell of the grass.jupyter notebook:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/tmp/ipykernel_100/570641186.py in <module>
----> 1 img.render(elevation_map="elevation", color_map="landuse", perspective=20)
      2 img.show()

~/dist.x86_64-pc-linux-gnu/etc/python/grass/jupyter/render3d.py in render(self, **kwargs)
    205                     env = os.environ.copy()
    206                 self._region_manager.set_region_from_command(env=env, **kwargs)
--> 207                 self.overlay.region_manager.set_region_from_env(env)
    208                 gs.run_command(module, env=env, **kwargs)
    209         else:

~/dist.x86_64-pc-linux-gnu/etc/python/grass/jupyter/region.py in set_region_from_env(self, env)
    125         """Copies GRASS_REGION from provided environment
    126         to local environment to set the computational region"""
--> 127         self._env["GRASS_REGION"] = env["GRASS_REGION"]
    128 
    129     def set_region_from_command(self, module, **kwargs):

KeyError: 'GRASS_REGION'
```

**Comment by wenzeslaus** [comment]
> I don't know if related, but I get the following error in the last cell of the grass.jupyter notebook: `...KeyError: 'GRASS_REGION'`

Hm, I can reproduce it, but it is unrelated to this PR. Likely coming from #1871.

**Review by chaedri — APPROVED** [review]
Code and description looks good to me. I ran both binder links (and looked at the other notebooks) and they all look good. This is a nice addition - it certainly makes the launch clearer.

**Comment by neteler** [comment]
@wenzeslaus do we want this backported to 8.0.0?

**Comment by wenzeslaus** [comment]
Technically, this missed the window for new features, but on the other hand, it is kind of a cap stone for many changes which are in 8.0 and it updates and fixes the documentation.

**Comment by neteler** [comment]
> Technically, this missed the window for new features, but on the other hand, it is kind of a cap stone for many changes which are in 8.0 and it updates and fixes the documentation.

Done. Cherry-picking went smoothly.

**Pull request merged**